packages/core/src/modules/connections/ConnectionsModule.ts

```diff
@@ -38,19 +43,59 @@ export class ConnectionsModule {
     agentConfig: AgentConfig,
     didExchangeProtocol: DidExchangeProtocol,
     connectionService: ConnectionService,
+    outOfBandService: OutOfBandService,
     trustPingService: TrustPingService,
     mediationRecipientService: MediationRecipientService,
     messageSender: MessageSender
   ) {
     this.agentConfig = agentConfig
     this.didExchangeProtocol = didExchangeProtocol
     this.connectionService = connectionService
+    this.outOfBandService = outOfBandService
     this.trustPingService = trustPingService
     this.mediationRecipientService = mediationRecipientService
     this.messageSender = messageSender
     this.registerHandlers(dispatcher)
   }
 
+  public async acceptOutOfBandInvitation(
+    outOfBandRecord: OutOfBandRecord,
+    config: {
+      autoAcceptConnection?: boolean
+      label?: string
+      mediatorId?: string
+      protocol: HandshakeProtocol
+    }
+  ) {
+    const { protocol, label, autoAcceptConnection } = config
+    const routing = await this.mediationRecipientService.getRouting({ mediatorId: config?.mediatorId })
+
+    let result
+    if (protocol === HandshakeProtocol.DidExchange) {
+      result = await this.didExchangeProtocol.createRequest(outOfBandRecord, {
+        label: label || this.agentConfig.label,
+        routing,
+        autoAcceptConnection,
+      })
+    } else if (protocol === HandshakeProtocol.Connections) {
+      result = await this.connectionService.protocolCreateRequest(outOfBandRecord, {
```

Comment:
What's the difference in naming of `createRequest` and `protocolCreateRequest`?

Reply:
The old one is to preserve the implementation relying on an existing connection record. It'll be removed as we discussed on the last AFJ call.